### PR TITLE
Update MSC2918 refresh token support to confirm with the latest revision: accept the `refresh_tokens` parameter in the request body rather than in the URL parameters.

### PR DESCRIPTION
**Pull request by reivilibre**
The latest revision of [MSC2918](https://github.com/matrix-org/matrix-doc/blob/main/proposals/2918-refreshtokens.md#msc2918-refresh-tokens) does it this way; previously a URL parameter was used.

### CODE REVIEW

File: synapse/rest/client/login.py [hunk 1]
```diff
@@ -155,11 +154,15 @@ async def on_POST(self, request: SynapseRequest) -> Tuple[int, LoginResponse]:
         login_submission = parse_json_object_from_request(request)
 
         if self._msc2918_enabled:
-            # Check if this login should also issue a refresh token, as per
-            # MSC2918
-            should_issue_refresh_token = parse_boolean(
-                request, name=LoginRestServlet.REFRESH_TOKEN_PARAM, default=False
+            # Check if this registration should also issue a refresh token, as
```
Comment (DMRobertson) [inline synapse/rest/client/login.py:157]: Why login -> registration?

**Review by DMRobertson — COMMENTED** [review]
One typo? Otherwise seems perfectly reasonable

File: synapse/rest/client/login.py (hunk 1 above)
Comment (reivilibre) [reply synapse/rest/client/login.py:157]: oops, good spot

**Review by DMRobertson — APPROVED** [review]
LGTM. :ship: :it: !

**Pull request merged**